Entities/Material.h

```diff
@@ -179,8 +185,10 @@ namespace RTE {
 		Color m_Color; //!< The natural color of this material.
 		bool m_UseOwnColor; //!< Whether or not to use the own color when a pixel of this material is knocked loose from the terrain. If 0, then the terrain pixel's color will be applied instead.
 
-		ContentFile m_TextureFile; //!< The file pointing to the terrain texture of this Material 
-		BITMAP *m_TerrainTexture; //!< The texture of this material, used when building an SLTerrain.
+		ContentFile m_FGTextureFile; //!< The file pointing to the terrain foreground texture of this Material.
+		ContentFile m_BGTextureFile; //!< The file pointing to the terrain background texture of this Material.
+		BITMAP *m_TerrainFGTexture; //!< The foreground texture of this Material, used when building an SLTerrain. Not owned.
```

Comment:
Why waste cycles every time you need to get the bitmap?
Getting it from the static map is infinitely more complex than storing the pointer.
Sure, you don't get these very often but it doesn't change the fact.

Reply:
Clean up the header a bit. As you say, I guess they're not gotten often so the cpu cycles wasted would be negligible (maybe, I haven't looked at this code in so long so I have no clue how much they're used lol).
Theoretically it'd speed up object creation, but that'd also be negligible too haha

Reply:
Nah, should always store a pointer to the bitmap.
Getting from static map is 900000000000000% slower because some evaluation and then lookup (fast one but still infinitely slower).
Having some extra clutter for gainzz is worth.

Reply:
I mean, again, it depends how often that access happens. If it happens once on scene load and adds 0.5 pico-seconds to scene load it's pretty silly. If it happens every frame or every n frames, that might be a different story.
I know you know enough now to know about premature optimization, even if you love the picogainz haha

Reply:
There is no premature in a 20+ year old codebase lel
Besides, some picogainzz lost here some nanogainzz lost there and bam we lost a whole gigagainzz.

Reply:
:me: